client/data/domains/use-get-domains-query.ts

```diff
@@ -9,7 +9,7 @@ type UseGetDomainsQueryData = ResponseDomain[];
 
 export const useGetDomainsQuery = (
 	siteId: number | null,
-	queryOptions?: UseQueryOptions< any, unknown, UseGetDomainsQueryData >
+	queryOptions?: Omit< UseQueryOptions< any, unknown, UseGetDomainsQueryData >, 'queryKey' >
```

Comment:
Do we want to start using the `Error` for the error type while we're here?

Same comment applies throughout the rest of the PR.

Reply:
Generally I think you're right. In #84474 I tried to only change occurrences where there was a mismatch due to the update and not touch too many of the usages to keep changes small and contained. There's a bunch of potential to refactor those to either not use Generics at all (and let RQ infer) or improve types where we do use Generics.

Reply:
Relying on inference would be ideal! Thanks!